addons/mail/models/mail_thread.py

```diff
@@ -835,10 +800,7 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
                 obj = self.env[alias.alias_parent_model_id.model].browse(alias.alias_parent_thread_id)
             else:
                 obj = self.env[model]
-            if hasattr(obj, '_alias_check_contact'):
-                error_code = obj._alias_check_contact(message, message_dict, alias)
-            else:
-                error_code = self.env['mail.alias.mixin']._alias_check_contact_on_record(obj, message, message_dict, alias)
+            error_code = obj._alias_check_contact(message, message_dict, alias)
             if error_code is not True:
```

Comment:
I lack a bit of context but there seems to be a mismatch of naming here.
"_alias_check_contact", returning an "error_code" (wow), that is ignored if... True? (wow²).

Suggestion:
"_get_alias_access_error" returning an error string (as currently) or False if no error occured.
And you would check "if error_message" here.
Makes a bit more sense, don't you think?
(Still not a fan of my naming proposition but maybe you will have better).